genkeymap/dump-keymaps.sh

```diff
@@ -1,11 +1,29 @@
 #!/bin/sh
 
-if ! command -v setxkbmap >/dev/null
-then
-  echo "error, setxkbmap not found"
-  exit 1
+# Run the keymap extraction in a clean X session
+if [ "$1" != _IN_XVFB_SESSION_ ]; then
+    # All commands available?
+    ok=1
+    for cmd in setxkbmap xvfb-run xauth; do
+        if ! command -v $cmd >/dev/null
+        then
+          echo "Error. $cmd not found" >&2
+          ok=
+        fi
+    done
+    if [ -z "$ok" ]; then
+        exit 1
+    fi
+
+    xvfb-run --auto-servernum $0 _IN_XVFB_SESSION_
```

Comment:
Just done some more testing, and I've found on my system that the X server is resetting between the `setxkbmap` command and the `xrdp-rungenkeymap` command.

This means that (for example) there's no difference between the generated file `km-00000409.ini` (US English) and `km-00000809.ini` (UK English). 

I think this line needs to be 

```
xvfb-run --auto-servernum --server-args="-noreset" $0 _IN_XVFB_SESSION_
```

Reply:
BTW: Apologies for introducing this bug.

Reply:
Thank you, fixed.